src/main/java/uk/gov/register/core/TransactionalMemoizationStoreImpl.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.register.core;
+
+import uk.gov.verifiablelog.store.memoization.MemoizationStore;
+import java.util.HashMap;
+
+public class TransactionalMemoizationStoreImpl implements TransactionalMemoizationStore {
+    private final MemoizationStore instance;
+    private final HashMap<Integer, HashMap<Integer, byte[]>> temporaryEntries;
+    private Integer currentEntryCount;
+
+    public TransactionalMemoizationStoreImpl(
+            MemoizationStore instance) {
+        this.instance = instance;
+        this.temporaryEntries = new HashMap<>();
+    }
+
+    @Override
+    public void setCurrentEntryCount(int currentEntryCount) {
+        this.currentEntryCount = currentEntryCount;
+    }
+
+    @Override
+    public void put(Integer start, Integer size, byte[] value) {
+        if ((start + size) <= currentEntryCount) {
+            return;
```

Comment:
shouldn't this pass those to the underlying store just like in `get`?
```if ((start + size) <= currentEntryCount) {
    instance.put(start, size, value);
    return;
}```
somink like that, what do you think?